src/proj.h

```diff
@@ -341,6 +341,9 @@ void proj_errno_restore (PJ *P, int err);
 
 char *proj_get_definition (PJ *P);
 
+/* deallocate anything obtained from PROJ.4 */
```

Comment:
So, @rouault  and @kbevers : What's to be done here? I seriously dislike the first solution: While `proj_get_definition()` /  `proj_buffer_free()` is a fairly mindless 1:1 translation from the original `pj_get_def()` / `pj_dalloc()` implementation, it is totally orthogonal to current semantics of the proj_xxx API.

I can think of the following solutions:

1. **As-is:** A create/destroy pair, that clearly signals that what's created must be destroyed - when you start a paranthesis, you must also provide a corresponding end paranthesis at the proper syntactic and semantic point. *I evidently much prefer this solution.*

2. **As-was:** The original `proj_get_definition()` / `proj_buffer_free()` pair, where the constructor/destructor pair is referentially asymmetric, where the constructor is specific, while the destructor is generic, and where, to add insult to injury, the naming is orthogonal to everything else in the proj_xxx API. *I very much dislike this solution.*

3. **As-maybe-to-be:** A not-really-getter-but-not-either-referentially-a-getter: We could introduce a `proj_definition()` function, which would be referentially in alignment with the `proj_error()` function, but would need a low level deallocator, which would be referentially assymetric to the constructor/pseudogetter, but which, on the positive side, could be play the role of a C++ default-destructor. In acknowledgement of the humble low-level building bricks of PROJ.4, I would prefer to call this function `proj_dealloc()`. *I could certainly live reasonably happily with this solution,* although I would very much prefer to make it more clear when something is constructed for the API user, and hence also should be destroyed by the API user.

There is probably at least half-a-dozen additional potential solutions, that I cannot think of right now - but have a go: Suggestions? Opinions?